### PR TITLE
DiscordPresence: use game-specific artwork if available

### PR DESCRIPTION
**Pull request by delroth**
image

Since we don't have a way (AFAIK) to dynamically collect the list of
available art assets, we hardcode a list of gameids with available
artwork inside Dolphin. It's not great, but I don't think it's a
terrible solution either.

Art has to be manually uploaded to our Discord app configuration, and we
have a limit of ~150 assets, so most likely we'll limit ourselves to a
small set of popular games.

### CODE REVIEW

File: Source/Core/UICommon/DiscordPresence.cpp [hunk 1]
```diff
@@ -82,6 +87,30 @@ static void HandleDiscordJoin(const char* join_secret)
...
+std::string ArtworkForGameId(const std::string& gameid)
+{
+  static const std::set<std::string> REGISTERED_GAMES{
+      "GALX01",
+  };
+
+  std::string region_neutral_gameid = gameid;
+  if (region_neutral_gameid.size() >= 4)
+  {
+    region_neutral_gameid[3] = 'X';
```
Comment (JosJuice) [inline Source/Core/UICommon/DiscordPresence.cpp]: The two last characters of a six-character game ID vary between regions in some cases (when a game has been published by different companies in different regions). If you want to make a game ID region neutral, I would suggest removing all characters except for the first three, like we do with game INIs.

File: Source/Core/UICommon/DiscordPresence.cpp [hunk 2]
```diff
@@ -82,6 +87,30 @@ static void HandleDiscordJoin(const char* join_secret)
...
+  }
+
+  if (REGISTERED_GAMES.count(region_neutral_gameid) != 0)
```
Comment (BhaaLseN) [inline Source/Core/UICommon/DiscordPresence.cpp]: What about `REGISTERED_GAMES.contains(region_neutral_gameid)`?

Comment (delroth) [reply Source/Core/UICommon/DiscordPresence.cpp]: That doesn't exist yet :)

File: Source/Core/UICommon/DiscordPresence.cpp (hunk 1 above)
Comment (delroth) [reply Source/Core/UICommon/DiscordPresence.cpp]: Fixed.

**Pull request merged**